File: 序号/350.两个数组的交集II.cpp
class Solution {
 public:
  // vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
  //   sort(nums1.begin(), nums1.end());
  //   sort(nums2.begin(), nums2.end());
  //   vector<int> v;
  //   set_intersection(nums1.begin(), nums1.end(), nums2.begin(), nums2.end(),
  //                    back_insert_iterator(v));
  //   return v;
  // }
  vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
    sort(nums1.begin(), nums1.end());
    sort(nums2.begin(), nums2.end());
    vector<int> v;
    int m = nums1.size(), n = nums2.size(), i = 0, j = 0;
    while (i < m && j < n) {
      if (nums1[i] < nums2[j]) {
        i++;
        continue;
      }
      if (nums1[i] > nums2[j]) {
        j++;
        continue;
      }
      v.emplace_back(nums1[i]);
      i++;
      j++;
    }
    return v;
  }
};

File: 序号/350. 两个数组的交集 II.py
'''
25/01/30
'''
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        m,n = len(nums1),len(nums2)
        nums1.sort()
        nums2.sort()
        i=j=0
        ans = []
        while i<m and j<n:
            if nums1[i]==nums2[j]:
                ans.append(nums1[i])
                i+=1
                j+=1
            elif nums1[i]>nums2[j]:
                j+=1
            else:
                i+=1
        return ans
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1.sort()
        nums2.sort()
        m,n=len(nums1),len(nums2)
        i,j=0,0
        ans=[]
        while i<m and j<n:
            if nums1[i]<nums2[j]:
                i+=1
                continue
            if nums1[i]>nums2[j]:
                j+=1
                continue
            ans.append(nums1[i])
            i+=1
            j+=1
        return ans